packages/aws-rfdk/lib/core/test/x509-certificate.test.ts

```diff
@@ -324,10 +325,87 @@ test('Generate cert, all options set', () => {
       ],
     },
   }));
+  expectCDK(stack).to(haveResourceLike('AWS::IAM::Policy', {
+    PolicyDocument: {
+      Statement: [
+        {
+          Action: 'secretsmanager:GetSecretValue',
+          Condition: {
+            StringEquals: {
+              'secretsmanager:ResourceTag/X509SecretGrant-B2B09A60': 'b2b09a6086e87fe14005f4e0b800e4f0',
+            },
+          },
+        },
+        {
+          Action: 'secretsmanager:ListSecrets',
+        },
+        {
+          Action: 'cloudwatch:PutMetricData',
+          Condition: {
+            StringEquals: {
+              'cloudwatch:namespace': CertificateRotationMonitor.METRIC_NAMESPACE,
+            },
+          },
+        },
+      ],
+    },
+  }));
   // Expect Lambda for doing the cert generation to use the generate() handler
   expectCDK(stack).to(haveResourceLike('AWS::Lambda::Function', {
     Handler: 'x509-certificate.generate',
   }));
+  // Expect Lambda for validating the cert experation period
+  expectCDK(stack).to(haveResourceLike('AWS::Lambda::Function', {
+    Handler: 'cert-rotation-monitor.handler',
+  }));
+  expectCDK(stack).to(haveResourceLike('AWS::CloudWatch::Alarm', {
+    ComparisonOperator: 'LessThanThreshold',
+    EvaluationPeriods: 1,
+    Dimensions: [
+      {
+        Name: CertificateRotationMonitor.METRIC_DIMENSION,
+        Value: 'b2b09a6086e87fe14005f4e0b800e4f0',
+      },
+    ],
+    MetricName: CertificateRotationMonitor.METRIC_NAME,
+    Namespace: CertificateRotationMonitor.METRIC_NAMESPACE,
+    Period: 86400,
+    TreatMissingData: 'notBreaching',
+    Threshold: 15,
+    Statistic: 'Minimum',
+    ActionsEnabled: true,
+  }));
+  expectCDK(stack).to(haveResourceLike('AWS::SNS::Subscription', {
+    Protocol: 'email',
+    Endpoint: 'test@amazon.com',
+  }));
+});
+
+test('Certificate alarm without action', () => {
+  const stack = new Stack();
+  const subject = { cn: 'testCN' };
+
+  new X509CertificatePem(stack, 'Cert', {
+    subject,
+  });
+  expectCDK(stack).to(haveResourceLike('AWS::CloudWatch::Alarm', {
+    ComparisonOperator: 'LessThanThreshold',
+    EvaluationPeriods: 1,
+    Dimensions: [
+      {
+        Name: CertificateRotationMonitor.METRIC_DIMENSION,
+        Value: 'b2b09a6086e87fe14005f4e0b800e4f0',
+      },
+    ],
+    MetricName: CertificateRotationMonitor.METRIC_NAME,
+    Namespace: CertificateRotationMonitor.METRIC_NAMESPACE,
+    Period: 86400,
+    TreatMissingData: 'notBreaching',
```

Comment:
should we alarm on missing data? that means that our Lambda is not publishing metrics correctly, right?